backend/src/team-events-imageAPI.ts

```diff
@@ -0,0 +1,73 @@
+import { bucket } from './firebase';
+import { getNetIDFromEmail } from './util';
+import { NotFoundError } from './errors';
+
+export const setProofImage = async (user: IdolMember, name: string): Promise<string> => {
+  const netId: string = getNetIDFromEmail(user.email);
+  const file = bucket.file(`eventProofs/${netId}/${name}.jpg`);
+  const signedURL = await file.getSignedUrl({
+    action: 'write',
+    version: 'v4',
+    expires: Date.now() + 15 * 60000 // 15 min
+  });
+  return signedURL[0];
+};
+
+export const getProofImage = async (user: IdolMember, name: string): Promise<string> => {
+  const netId: string = getNetIDFromEmail(user.email);
+  const file = bucket.file(`eventProofs/${netId}/${name}.jpg`);
+  const fileExists = await file.exists().then((result) => result[0]);
+  if (!fileExists) {
+    throw new NotFoundError(`The requested image (${netId}/${name}.jpg) does not exist`);
+  }
+  const signedUrl = await file.getSignedUrl({
+    action: 'read',
+    expires: Date.now() + 15 * 60000
+  });
+  return signedUrl[0];
+};
+
+export const allProofImagesForMember = async (
+  user: IdolMember
+): Promise<readonly EventProofImage[]> => {
+  const netId: string = getNetIDFromEmail(user.email);
+  const files = await bucket.getFiles({ prefix: `eventProofs/${netId}` });
+  const images = await Promise.all(
+    files[0].map(async (file) => {
+      const signedURL = await file.getSignedUrl({
+        action: 'read',
+        expires: Date.now() + 15 * 60000 // 15 min
+      });
+      const fileName = await file.getMetadata().then((data) => data[1].body.name);
+      return {
+        fileName,
+        url: signedURL[0]
+      };
+    })
+  );
+
+  images
+    .filter((image) => image.fileName.length > 'eventProofs/'.length)
+    .map((image) => ({
+      ...image,
+      fileName: image.fileName.slice(image.fileName.indexOf('/') + 1)
+    }));
+
+  return images;
+};
+
+export const deleteProofImage = async (user: IdolMember, name: string): Promise<void> => {
+  const netId: string = getNetIDFromEmail(user.email);
+  const imageFile = bucket.file(`eventProofs/${netId}/${name}.jpg`);
```

Comment:
Should there be a quick check to make sure the file exists first? I'm never sure what the best practice is since if the file doesn't exist, then the outcome of "deleteProofImage" is the same.

Reply:
Should be fine without the check, the purpose of deleting the proof image is just to save memory. From my other comment, proof images should only be deleted if request if approved/rejected. If it exists then we get rid of it, if it doesn't then it's also fine.